app/server.js

```diff
@@ -3,11 +3,269 @@ var fs = require('fs');
+  brands = JSON.parse(fs.readFileSync("initial-data/brands.json","utf-8"));
+
+  //populate products
+  products = JSON.parse(fs.readFileSync("initial-data/products.json","utf-8"));
+
+  //populate users
+  users = JSON.parse(fs.readFileSync("initial-data/users.json","utf-8"));
+  
+});
+
+//GET api/products
+router.get('/api/products', (request, response) => {
+  response.writeHead(200, { "Content-Type": "application/json" });
+  return response.end(JSON.stringify(products));
+});
+
+//GET api/brands
+router.get('/api/brands', (request, response) => {
+  response.writeHead(200, { "Content-Type": "application/json" });
+  return response.end(JSON.stringify(brands));
+});
+
+//GET api/brands/:brandId/products
+router.get('/api/brands/:brandId/products', (request, response) => {
+  const { brandId } = request.params;
+  const brandToFilterBy = brands.find(brand => brand.id == brandId);
+  if (!brandToFilterBy) {
+    response.writeHead(404, "There are no products associated with that brand ID");
+    return response.end();
+  }
+  
+  response.writeHead(200, { "Content-Type": "application/json" });
+  const filteredProducts = products.filter(
+    products => products.categoryId === brandToFilterBy.id
+  );
+  
+  return response.end(JSON.stringify(filteredProducts));
+});
+
+//POST api/login
+router.post('/api/login', (request, response) => {
+  if (request.body.username && request.body.password) {
+    let user = users.find((user) => {
+      return user.login.username == request.body.username && user.login.password == request.body.password;
+    });
+    if (user) {
+      response.writeHead(200, {'Content-Type': 'application/json'});
+      let currentAccessToken = accessTokens.find((tokenObject) => {
+        return tokenObject.username == user.login.username;
+      });
+
+      if (currentAccessToken) {
+        return response.end(JSON.stringify(currentAccessToken.token));
+
+      } else {
+        let newAccessToken = {
+          username: user.login.username,
+          token: uid(16)
+        }
+        accessTokens.push(newAccessToken);
+        return response.end(JSON.stringify(newAccessToken.token));
+      }
+
+    } else {
+      response.writeHead(401, "Invalid username or password");
+      return response.end();      
+    }
+} else {
+    response.writeHead(400, "Incorrectly formatted response");
+    return response.end();
+  }
+});
+
+//Helper function to verify tokens
+const verifyAccessToken = (tokenToVerify) => {
+  if (!tokenToVerify) {
+    return 401;
+  } 
+  let currentAccessToken = accessTokens.find(accessToken => {
+    return accessToken.token == tokenToVerify;
+  });
+  if (!currentAccessToken) {
+    return 403;
+  }
+  if (currentAccessToken) {
+    return currentAccessToken;
+  }
+};
+
+//GET api/me/cart
+router.get('/api/me/cart', (request, response) => {
+  const parsedUrl = require('url').parse(request.url, true);
+  let tokenVerified = verifyAccessToken(parsedUrl.query.accessToken);
```

Comment:
In a professional setting it sometimes makes more sense to pass the token as a header, even in a GET request. This works fine for our purposes here, just a heads up!